tests/compas/test_iotools.py

```diff
@@ -0,0 +1,125 @@
+import io
+import math
+import os
+import tempfile
+
+import pytest
+
+from compas import _iotools
+
+BASE_FOLDER = os.path.dirname(__file__)
+IMAGE_FILE_SIZE = 252391
+TEXT_FILE_SIZE = 747
+REMOTE_IMAGE_FILE_SIZE = 5430
+
+
+@pytest.fixture
+def path_image():
+    return os.path.join(BASE_FOLDER, 'fixtures', 'iotools', 'image.png')
+
+
+@pytest.fixture
+def path_text():
+    return os.path.join(BASE_FOLDER, 'fixtures', 'iotools', 'text.txt')
+
+
+@pytest.fixture
+def url_text():
+    return 'https://raw.githubusercontent.com/compas-dev/compas/master/README.md'
+
+
+@pytest.fixture
+def url_image():
+    return 'http://google.com/favicon.ico'
+
+
+def test_open_file_path_binary(path_image):
+    with _iotools.open_file(path_image, mode='rb') as file:
+        assert len(file.read()) == IMAGE_FILE_SIZE
+
+
+def test_open_file_path_text(path_text):
+    with _iotools.open_file(path_text, mode='r') as file:
+        assert len(file.read()) == TEXT_FILE_SIZE
+
+
+def test_open_file_object_binary(path_image):
+    with open(path_image, mode='rb') as f:
+        with _iotools.open_file(f) as file:
+            assert len(file.read()) == IMAGE_FILE_SIZE
+
+
+def test_open_file_object_text(path_text):
+    with open(path_text, mode='r') as f:
+        with _iotools.open_file(f) as file:
+            assert len(file.read()) == TEXT_FILE_SIZE
+
+
+def test_open_file_memory_stream():
+    text = b"How many roads must a man walk down?"
```

Comment:
Again, legality?  Especially since this was just sold....

Reply:
hehe..! changed to public domain and CreativeCommons image! ;) https://github.com/compas-dev/compas/pull/712/commits/7b15f1677a5c6f540b5a5b78ccab8c395ee03955